packages/core/src/modules/discover-features/FeatureRegistry.ts

```diff
@@ -0,0 +1,30 @@
+import type { Feature, FeatureQuery } from './models'
+
+import { injectable } from '../../plugins'
+
+@injectable()
+class FeatureRegistry {
+  private features: Feature[] = []
+
+  public register<F extends Feature = Feature>(feature: F) {
```

Comment:
```suggestion
  public register<F extends Feature>(feature: F) {
```

This does not seem required

Reply:
True. At first I was playing with interfaces because these "Features" are quite arbitrary and generic, but at the moment I'm using classes. Probably not the best approach though, because currently we need to provide instances and would like to use options instead.